src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -231,8 +233,10 @@ public WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventTypes,
         /// <returns>A task representing the operation of getting the structure that contains the result of the waiting operation.</returns>
         public ValueTask<WaitForEventResult> WaitForEventAsync(int pinNumber, PinEventTypes eventTypes, TimeSpan timeout)
         {
-            CancellationTokenSource tokenSource = new CancellationTokenSource(timeout);
-            return WaitForEventAsync(pinNumber, eventTypes, tokenSource.Token);
+            using (CancellationTokenSource tokenSource = new CancellationTokenSource(timeout))
+            {
+                return WaitForEventAsync(pinNumber, eventTypes, tokenSource.Token);
```

Comment:
This one seems problematic without `await`.

Reply:
Sorry for the sloppy work. Will await for WaitForEventAsync. What about ConfigureAwait(false)?

Reply:
I'm not sure what the best practice is for this repository. It seems like the ideal behavior would match `ConfigureAwaitRunInline`, but I don't know how to achieve that in the most efficient manner.

Reply:
Implemented the changes, can you please take a look?